react/features/base/dialog/components/Dialog.web.js

```diff
@@ -41,28 +53,138 @@ class Dialog extends AbstractDialog {
     }
 
     /**
+     * Initializes a new Dialog instance.
+     *
+     * @param {Object} props - The read-only properties with which the new
+     * instance is to be initialized.
+     */
+    constructor(props) {
+        super(props);
+
+        this._onKeyDown = this._onKeyDown.bind(this);
+        this._setDialogElement = this._setDialogElement.bind(this);
+    }
+
+    /**
+     * React Component method that executes once component is mounted.
+     *
+     * @inheritdoc
+     */
+    componentDidMount() {
+        this.updateButtonFocus();
+
+        // adds listener for enter key
+        document.addEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * Clears the listener.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentWillUnmount() {
+        document.removeEventListener('keydown', this._onKeyDown);
+    }
+
+    /**
+     * React Component method that executes once component is updated.
+     *
+     * @inheritdoc
+     * @returns {void}
+     */
+    componentDidUpdate(prevProps) {
+        // if there is an update in any of the buttons enable/disable props
+        // update the focus if needed
+        if (prevProps.okDisabled !== this.props.okDisabled
```

Comment:
We only care whether a button visibility has changed (okDisabled or cancelDisabled). There is a case for dialout dialog which initially is not showing the ok button but later enables it after user input.